channels/api.py

```diff
@@ -471,7 +483,13 @@ def create_channel(
             **other_settings,
         )
 
-        Channel.objects.create(name=name, membership_is_managed=membership_is_managed)
+        # create an empty widget list for new channels
+        widget_list = WidgetList.objects.create()
```

Comment:
I agree, in fact, I've already addressed this in my [other PR for article work](https://github.com/mitodl/open-discussions/pull/1517/files#diff-6b4e079385252444280613805ab05474R515), because even with a transaction here, we can still end up in an inconsistent state because reddit doesn't get rolled back. Problem then, is you'll get an integrity error on the channel write that preempts the current reddit duplicate channel error we're relying on, so **that** then needs to get remapped.

Since I've already done all that work in that other PR I'd suggest either

a) deferring adding this transaction and just rebasing the other PR once this merges
b) block merging this PR until the other one is reviewed and merged 

Thoughts?